css-multicol-2/Overview.bs

```diff
@@ -360,11 +360,14 @@ The multi-column model</h2>
 
 	The column boxes are ordered
 	in the [=inline base direction=] of the multicol container
-	and arranged into <dfn lt="multi-column line | multi-col line | multicol line">multicol lines</dfn>.
+	and arranged into 
+	<dfn lt="multi-column row | multi-col row | multicol row">multicol rows</dfn>.
+	A multicol row contains at least one <dfn lt="multi-column line | multi-col line | multicol line">multicol line</dfn>,
+	but may contain additional multicol lines due to spanning elements.
 	The <dfn export>column width</dfn> is the length of the column box in the inline direction.
 	The <dfn export>column height</dfn> is the length of the column box in the block direction.
-	All column boxes in a line have the same column width,
-	and all column boxes in a line have the same column height.
+	All column boxes in a row have the same column width,
```

Comment:
Column boxes are inside a line (which is inside a row). Why this change?

Reply:
Assuming that we have the same amount of content before and after the spanner, then yes, the columns will all have a height of 40px, both before and after the spanner.

```html
<!DOCTYPE html>
<div style="columns:3; column-fill:auto; column-height:100px; line-height:20px;">
  before
  <div style="column-span:all;">spanner</div>
  after<br>after<br>after<br>after<br>after<br>
</div>
```

There's a multicol container. It has one row that is 100px tall. The row has one line, one spanner, then another line.
The first line is 20px tall, the spanner is 20px tall, the second line is 60px tall. The columns in the first line is 20px tall. The columns in the second line are 60px tall. Therefore all columns in a row do not necessarily have the same column height.

Reply:
I see what you mean... maybe we need to more strongly define column boxes vs columns? As it's the column that has the height, depending on how many lines there are multiple column boxes within each column. I think that's where I'm confusing things anyway.

Reply:
Ah, got it! I just read `column-height` out loud now, and got it. I must have been pronouncing it as "row height" all along. And that's what it is, isn't it? Column flow may be interrupted by a spanner, resulting in two column lines (one before, one after), and the total height available for columns will be `column-height` minus the space taken up by the spanner. Grouping column boxes with the same inline offset (before and after a spanner) together doesn't make sense to me.

Are we sure about the naming, then? Maybe it should be `row-height`, or `column-row-height`?

Reply:
Ok. Again I'm tempted to conflate "lines" and "rows", and let column spanners live on the outside of that, then define that row gaps and decorations are only inserted if there's column content on both sides of the gap (i.e. only when columns wrap, and not e.g. next to spanners or at the bottom / top when crossing an outer fragmentainer boundary). What would we lose with such a simplified model? `column-height` would be for columns only, then. Any spanner would be pushed after the row/line, take up its space, and a new row/line would resume after it. What have I forgotten about?

Reply:
That won't work for the carousel type use cases. You'd end up with a spanner outside of the "page" which would be weird. 

It would be fine for the simplest use case of just making sure people don't need to scroll up and down to read, but if we want an entire screen to be a row AND you had a spanner, you'd then end up with the spanner off the bottom of the screen.

Also it would definitely stop `column-span: <integer>` ever being a thing.

We need the lines.

Reply:
To explain about `column-span: <integer>` the most likely way (in my mind) to implement that would be in conjunction with page floats. So you'd have a three column multicol container, and an element spanning two columns floated top right. That's a super common magazine type layout and the sort of thing people raise (for example [#11474](https://github.com/w3c/csswg-drafts/issues/11474)).

That would not work if spanners were this weird thing that existed outside of rows.

Reply:
I think it would be such an unexpected behavior for spanners to do that, to immediately lose the nice "set of pages at x height" because there's a spanner seems bad. I do thing that a carousel type layout using page floats would likely be something people want if we ever go there, because there's some nice (and fairly obvious) patterns you could achieve with that.